File: src/main/java/it/polito/ai/pedibus/api/models/Reservation.java
package it.polito.ai.pedibus.api.models;


import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import it.polito.ai.pedibus.api.serializers.ObjectIdSerializer;
import lombok.Builder;
import lombok.Data;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;


@Data
@Builder
@Document(collection = "reservations")
public class Reservation {
    public enum Direction {OUTWARD, BACK}

    @Id
    @JsonSerialize(using = ObjectIdSerializer.class)
    private ObjectId id;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern="dd-MM-yyyy")
    private LocalDate date;
    private String lineName;
    // We're not sure an ID is needed here, since every stop name, in the context of a direction and tripIndex number, is unique.
    private String childName;
    private Direction direction;
    private String stopName;
    // This just represents an index in the array of trips for that line, in that direction.
    private Integer tripIndex;
}

File: frontend/src/app/models/reservation.ts
import {Child} from "./child";
import {IChildReservationInfo} from "./ichild-reservation-info";
import {LocalDate} from "js-joda";

export class Reservation {
    private _id: string;
    private _stopName: string;
    private _childs: IChildReservationInfo[];
    private _direction: string;
    private _tripIndex: number;
    private _date: LocalDate;
    private _color: string;
    private _lineName: string;
    private _booked: boolean;
    private _childId: string;

    get stopName(){
        return this._stopName;
    }

    set stopName(value) {
        this._stopName = value;
    }

    get childs(){
        return this._childs;
    }

    set childs(value) {
        this._childs= value;
    }

    get direction(){
        return this._direction;
    }

    set direction(value){
        this._direction = value;
    }

    get tripIndex(){
        return this._tripIndex;
    }

    set tripIndex(value){
        this._tripIndex = value;
    }

    get date(){
        return this._date;
    }

    set date(value){
        this._date = value;
    }

    get color(){
        return this._color;
    }

    set color(value){
        this._color = value;
    }

    get lineName(){
        return this._lineName;
    }

    set lineName(value){
        this._lineName = value;
    }

    get booked(){
        return this._booked;
    }

    set booked(value){
        this._booked = value;
    }

    get id(){
        return this._id
    }

    set id(value){
        this._id = value;
    }

    get childId(){
        return this._childId;
    }

    set childId(value){
        this._childId = value
    }
}


